src/services/lease-service/get-tenant.ts

```diff
@@ -0,0 +1,161 @@
+import * as tenantLeaseAdapter from './adapters/xpand/tenant-lease-adapter'
+import * as xpandSoapAdapter from './adapters/xpand/xpand-soap-adapter'
+import * as priorityListService from './priority-list-service'
+
+type GetTenantError =
+  | 'get-contact'
+  | 'contact-not-found'
+  | 'contact-not-tenant'
+  | 'get-waiting-lists'
+  | 'waiting-list-internal-parking-space-not-found'
+  | 'get-contact-leases'
+  | 'contact-leases-not-found'
+  | 'get-residential-area'
+  | 'housing-contracts-not-found'
+  | 'get-lease-property-info'
+
+export type Tenant = Omit<Contact, 'leases'> & {
+  isTenant: true
+  queuePoints: number
+  currentHousingContract?: Lease
+  upcomingHousingContract?: Lease
+  parkingSpaceContracts?: Lease[]
+}
+
+export async function getTenant(params: {
+  contactCode: string
+}): Promise<tenantLeaseAdapter.AdapterResult<Tenant, GetTenantError>> {
+  const contact = await tenantLeaseAdapter.getContactByContactCode(
+    params.contactCode,
+    'false'
+  )
+  if (!contact.ok) {
+    return { ok: false, err: 'get-contact' }
+  }
+
+  if (!contact.data) {
+    return { ok: false, err: 'contact-not-found' }
+  }
+
+  if (contact.data.isTenant !== true) {
+    return { ok: false, err: 'contact-not-tenant' }
+  }
+
+  const waitingList = await xpandSoapAdapter.getWaitingList(
+    contact.data.nationalRegistrationNumber
+  )
+
+  if (!waitingList.ok) {
+    return { ok: false, err: 'get-waiting-lists' }
+  }
+
+  const waitingListForInternalParkingSpace =
+    priorityListService.parseWaitingListForInternalParkingSpace(
+      waitingList.data
+    )
+
+  if (!waitingListForInternalParkingSpace) {
+    return {
+      ok: false,
+      err: 'waiting-list-internal-parking-space-not-found',
+    }
+  }
+
+  const leases = await tenantLeaseAdapter.getLeasesForContactCode(
+    contact.data.contactCode,
+    'true',
+    undefined
+  )
+
+  if (!leases.ok) {
+    return {
+      ok: false,
+      err: 'get-contact-leases',
+    }
+  }
+
+  if (!leases.data.length) {
+    return {
+      ok: false,
+      err: 'contact-leases-not-found',
+    }
+  }
+
+  const activeAndUpcomingLeases = leases.data.filter(
+    priorityListService.isLeaseActiveOrUpcoming
+  )
+
+  const leasesWithResidentialArea = await Promise.all(
+    activeAndUpcomingLeases.map(async (lease) => {
+      const residentialArea =
+        await tenantLeaseAdapter.getResidentialAreaByRentalPropertyId(
+          lease.rentalPropertyId
+        )
+
+      if (!residentialArea.ok) {
+        throw new Error('Err getting residential area')
+      }
+
+      return {
+        ...lease,
+        residentialArea: residentialArea.data,
+      }
+    })
+  )
+    .then((data) => ({ ok: true, data }) as const)
+    .catch((err) => ({ ok: false, err }) as const)
+
+  if (!leasesWithResidentialArea.ok) {
+    return { ok: false, err: 'get-residential-area' }
+  }
+
+  const housingContracts = priorityListService.parseLeasesForHousingContracts(
```

Comment:
A bit out of scope but the utility functions to parse leases and parking spaces could also be refactored to some util I think. Not needed for this PR though :)